clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEvent.java

```diff
@@ -25,7 +25,8 @@ public abstract class ApplicationEvent {
 
     public enum Type {
         COMMIT, POLL, FETCH_COMMITTED_OFFSET, METADATA_UPDATE, ASSIGNMENT_CHANGE,
-        LIST_OFFSETS, RESET_POSITIONS, VALIDATE_POSITIONS, TOPIC_METADATA
+        LIST_OFFSETS, RESET_POSITIONS, VALIDATE_POSITIONS, TOPIC_METADATA, SUBSCRIPTION_CHANGE,
```

Comment:
It's exactly when the user changes the subscription via a call to subscribe. I used the name `SUBSCRIPTION_CHANGE` because it seemed clear and to be consistent with the existing `ASSIGNMENT_CHANGE`, but let me know if you think another name would be better.